src/methods/angles.jl

```diff
@@ -0,0 +1,174 @@
+# # Angles
+export angles
+
+#=
+## What is angles?
+
+Angles are the angles formed by a given geometries line segments, if it has line segments.
+
+To provide an example, consider this rectangle:
+```@example angles
+using GeometryOps
+using GeometryOps.GeometryBasics
+using Makie
+
+rect = Polygon([Point(0,0), Point(0,1), Point(1,1), Point(1,0), Point(0, 0)])
+f, a, p = poly(rect; axis = (; aspect = DataAspect()))
+```
+This is clearly a rectangle, with angles of 90 degrees.
+```@example angles
+lines!(a, rect; color = 1:length(coordinates(rect))+1)
+f
+angles(area)
+```
+
+## Implementation
+
+This is the GeoInterface-compatible implementation. First, we implement a
+wrapper method that dispatches to the correct implementation based on the
+geometry trait. This is also used in the implementation, since it's a lot less
+work!
+=#
+
+const _ANGLE_TARGETS = Union{GI.PolygonTrait,GI.AbstractCurveTrait,GI.MultiPointTrait,GI.PointTrait}
+
+"""
+    angles(geom, ::Type{T} = Float64)
+
+Returns the angles of a geometry or collection of geometries. 
+This is computed differently for different geometries:
+
+    - The angles of a point is an empty vector.
+    - The angles of a single line segment is an empty vector.
+    - The angles of a linestring or linearring is a vector of angles formed by the curve.
+    - The angles of a polygin is a vector of vectors of angles formed by each ring.
+    - The angles of a multi-geometry collection is a vector of the angles of each of the
+        sub-geometries as defined above.
+
+Result will be a Vector, or nested set of vectors, of type T where an optional argument with
+a default value of Float64.
+"""
+function angles(geom, ::Type{T} = Float64; threaded =false) where T <: AbstractFloat
+    applyreduce(vcat, _ANGLE_TARGETS, geom; threaded, init = Vector{T}()) do g
+        _angles(T, GI.trait(g), g)
+    end
+end
+
+# Points and single line segments have no angles
+_angles(::Type{T}, ::Union{GI.PointTrait, GI.LineTrait}, geom) where T = T[]
```

Comment:
Maybe MultiPointTrait is missing here?
```suggestion
_angles(::Type{T}, ::Union{GI.PointTrait, GI.LineTrait, GI.MultiPointTrait}, geom) where T = T[]
```